prometheus_hardware_exporter/collectors/redfish.py

```diff
@@ -1,24 +1,49 @@
 """Redfish collector."""
 from logging import getLogger
-from typing import Any, Dict, List, Optional
+from typing import Any, Callable, Dict, List, Optional
 
 import redfish
 import redfish_utilities
-from redfish.rest.v1 import HttpClient, InvalidCredentialsError, SessionCreationError
+from cachetools.func import ttl_cache
+from redfish.rest.v1 import (
+    HttpClient,
+    InvalidCredentialsError,
+    RetriesExhaustedError,
+    SessionCreationError,
+)
 
 logger = getLogger(__name__)
 
 
+# pylint: disable=R0903
+# pylint: disable=R0913
+
+
 class RedfishHelper:
     """Helper function for redfish."""
 
-    def get_sensor_data(self, host: str, username: str, password: str) -> Dict[str, List]:
+    def __init__(self, discover_cache_ttl: int) -> None:
+        """Initialize disover method with TTL value."""
+        self.discover = self.get_discover(discover_cache_ttl)
```

Comment:
Why don't we initialize the helper with host: str, username: str, password: str, timeout: int, max_retry: int

Reply:
Or just with Config

Reply:
That's a really nice idea. We wouldn't need to pass these login parameters each time to the get_sensor methods and while adding new methods for additional redfish metrics, their arg list would be simpler too. Will implement and push out this change :100: